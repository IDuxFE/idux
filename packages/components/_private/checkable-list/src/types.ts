/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
