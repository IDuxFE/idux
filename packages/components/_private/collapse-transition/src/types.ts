/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
