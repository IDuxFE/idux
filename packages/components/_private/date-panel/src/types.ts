/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export const datePanelProps = {
  cellTooltip: Function as PropType<(cell: { value: Date; disabled: boolean }) => string | void>,
  disabledDate: Function as PropType<(date: Date) => boolean>,
  type: {
    type: String as PropType<DatePanelType>,
    default: 'date',
  },
  value: [Date, Array] as PropType<Date | (Date | undefined)[]>,
  activeDate: Date,
  visible: {
    type: Boolean,
    default: undefined,
  },
  isSelecting: {
    type: Boolean,
    default: undefined,
  },

  onCellClick: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  onCellMouseenter: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  'onUpdate:activeDate': [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
}

export type DatePanelProps = ExtractInnerPropTypes<typeof datePanelProps>
export type DatePanelPublicProps = ExtractPublicPropTypes<typeof datePanelProps>
export type DatePanelComponent = DefineComponent<
  Omit<HTMLAttributes, keyof DatePanelPublicProps> & DatePanelPublicProps
>
export type DatePanelInstance = InstanceType<DefineComponent<DatePanelProps>>

export type DatePanelType = 'date' | 'week' | 'month' | 'quarter' | 'year'

// private
export const panelRowProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
}

export const panelCellProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
  cellIndex: {
    type: Number,
    required: true,
  },
}
