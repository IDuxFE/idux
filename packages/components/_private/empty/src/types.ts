/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { EmptyProps as IxEmptyProps } from '@idux/components/empty'

export interface EmptyProps {
  empty?: string | IxEmptyProps
}
