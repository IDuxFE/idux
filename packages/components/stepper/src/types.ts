/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { type DefineComponent, type HTMLAttributes } from 'vue'

import { type IxInnerPropTypes, IxPropTypes, type IxPublicPropTypes, type VKey } from '@idux/cdk/utils'

export type StepperLabelPlacement = 'end' | 'bottom'
export type StepperSize = 'md' | 'sm'
export type StepperStatus = 'process' | 'finish' | 'wait' | 'error'

export const stepperProps = {
  activeKey: IxPropTypes.oneOfType<VKey>([String, Number, Symbol]),
  clickable: IxPropTypes.bool,
  labelPlacement: IxPropTypes.oneOf<StepperLabelPlacement>(['end', 'bottom']),
  percent: IxPropTypes.range(0, 100),
  size: IxPropTypes.oneOf<StepperSize>(['md', 'sm']),
  status: IxPropTypes.oneOf<StepperStatus>(['process', 'finish', 'wait', 'error']).def('process'),
  vertical: IxPropTypes.bool.def(false),

  'onUpdate:activeKey': IxPropTypes.emit<(key: VKey) => void>(),
}

export type StepperProps = IxInnerPropTypes<typeof stepperProps>
export type StepperPublicProps = IxPublicPropTypes<typeof stepperProps>
export type StepperComponent = DefineComponent<Omit<HTMLAttributes, keyof StepperPublicProps> & StepperPublicProps>
export type StepperInstance = InstanceType<DefineComponent<StepperProps>>

export const stepperItemProps = {
  description: IxPropTypes.string,
  disabled: IxPropTypes.bool.def(false),
  icon: IxPropTypes.string,
  title: IxPropTypes.string,
  status: IxPropTypes.oneOf<StepperStatus>(['process', 'finish', 'wait', 'error']),
}

export type StepperItemProps = IxInnerPropTypes<typeof stepperItemProps>
export type StepperItemPublicProps = IxPublicPropTypes<typeof stepperItemProps>
export type StepperItemComponent = DefineComponent<
  Omit<HTMLAttributes, keyof StepperItemPublicProps> & StepperItemPublicProps
>
export type StepperItemInstance = InstanceType<DefineComponent<StepperItemProps>>
