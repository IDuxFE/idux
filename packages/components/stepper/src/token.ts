/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { type ComputedRef, type InjectionKey, type Slots } from 'vue'

import { type VKey } from '@idux/cdk/utils'

import { type StepperProps } from './types'

export interface StepperContext {
  props: StepperProps
  slots: Slots
  activeKey: ComputedRef<VKey | undefined>
  setActiveKey: (value: VKey) => void
}

export const stepperToken: InjectionKey<StepperContext> = Symbol('stepperToken')

export const stepperItemKey = Symbol('IDUX_STEPPER_ITEM_KEY')
