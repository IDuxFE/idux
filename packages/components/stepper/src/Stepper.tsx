/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { computed, defineComponent, normalizeClass, provide } from 'vue'

import { flattenNode, useControlledProp } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'

import { stepperItemKey, stepperToken } from './token'
import { stepperProps } from './types'

export default defineComponent({
  name: 'IxStepper',
  props: stepperProps,
  setup(props, { slots }) {
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-stepper`)
    const config = useGlobalConfig('stepper')

    const classes = computed(() => {
      const prefixCls = mergedPrefixCls.value
      const { size = config.size, labelPlacement = config.labelPlacement, percent, vertical } = props
      return normalizeClass({
        [prefixCls]: true,
        [`${prefixCls}-label-${labelPlacement}`]: true,
        [`${prefixCls}-${size}`]: true,
        [`${prefixCls}-vertical`]: vertical,
        [`${prefixCls}-with-percent`]: percent != null,
      })
    })

    const [activeKey, setActiveKey] = useControlledProp(props, 'activeKey')

    provide(stepperToken, { props, slots, activeKey, setActiveKey })

    return () => {
      const children = flattenNode(slots.default?.(), { key: stepperItemKey }).map((item, index) => {
        if (item.key == null) {
          item.key = index + 1
        }
        return item
      })
      return <div class={classes.value}>{children}</div>
    }
  },
})
