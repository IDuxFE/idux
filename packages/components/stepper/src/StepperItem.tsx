/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { type ComputedRef, type Slots, VNodeChild, computed, defineComponent, inject, normalizeClass } from 'vue'

import { isSymbol } from 'lodash-es'

import { NoopFunction, type VKey } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { IxIcon } from '@idux/components/icon'
import { IxProgress } from '@idux/components/progress'
import { covertStringVNode, useKey } from '@idux/components/utils'

import { stepperItemKey, stepperToken } from './token'
import { type StepperItemProps, type StepperProps, type StepperStatus, stepperItemProps } from './types'

export default defineComponent({
  name: 'IxStepperItem',
  props: stepperItemProps,
  [stepperItemKey]: true,
  setup(props, { slots }) {
    const key = useKey()
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-stepper-item`)
    // stepsContext must exist
    const { props: parentProps, activeKey, setActiveKey } = inject(stepperToken)!
    const isActive = computed(() => activeKey.value === key)
    const status = computed(() => {
      if (props.status) {
        return props.status
      }
      if (isActive.value) {
        return parentProps.status
      }
      const currActiveKey = activeKey.value
      if (!currActiveKey || isSymbol(currActiveKey) || isSymbol(key)) {
        return 'wait'
      }
      return currActiveKey > key ? 'finish' : 'wait'
    })

    const classes = computed(() => {
      const prefixCls = mergedPrefixCls.value
      const { disabled, icon } = props
      return normalizeClass({
        [prefixCls]: true,
        [`${prefixCls}-${status.value}`]: true,
        [`${prefixCls}-active`]: isActive.value,
        [`${prefixCls}-clickable`]: parentProps.clickable && !disabled,
        [`${prefixCls}-disabled`]: disabled,
        [`${prefixCls}-with-icon`]: icon || !!slots.icon,
      })
    })

    const onClick = () => {
      if (isActive.value) {
        return
      }
      setActiveKey(key)
    }

    return () => {
      const prefixCls = mergedPrefixCls.value
      const clickable = parentProps.clickable && !props.disabled
      const iconNode = renderIcon(props, slots, parentProps, status, key)
      const titleNode = covertStringVNode(slots, props, 'title')
      const descriptionNode = covertStringVNode(slots, props, 'description')
      return (
        <div
          class={classes.value}
          role={clickable ? 'button' : undefined}
          tabindex={clickable ? 0 : undefined}
          onClick={clickable ? onClick : undefined}
        >
          <div class={`${prefixCls}-tail`}></div>
          <div class={`${prefixCls}-icon`}>{iconNode}</div>
          <div class={`${prefixCls}-content`}>
            <div class={`${prefixCls}-title`}>{titleNode}</div>
            {descriptionNode && <div class={`${prefixCls}-description`}>{descriptionNode}</div>}
          </div>
        </div>
      )
    }
  },
})

const statusIcons: Partial<Record<StepperStatus, string>> = {
  finish: 'check',
  error: 'close',
}

function renderIcon(
  props: StepperItemProps,
  slots: Slots,
  parentProps: StepperProps,
  status: ComputedRef<StepperStatus>,
  key: VKey,
) {
  const currStatus = status.value
  let node: VNodeChild
  if (slots.icon) {
    node = slots.icon({ key, status: currStatus })
  } else {
    const { icon = statusIcons[currStatus] } = props
    if (icon) {
      node = <IxIcon name={icon} />
    } else {
      node = <span>{key}</span>
    }
  }

  if (currStatus === 'process' && parentProps.percent != null) {
    const width = parentProps.size === 'sm' ? 32 : 40
    return [
      <IxProgress type="circle" percent={parentProps.percent} width={width} strokeWidth={4} format={NoopFunction} />,
      node,
    ]
  }

  return node
}
