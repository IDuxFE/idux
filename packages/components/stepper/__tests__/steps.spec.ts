import { MountingOptions, mount } from '@vue/test-utils'
import { h } from 'vue'

import { renderWork } from '@tests'

import IxStepper from '../src/Stepper'
import IxStepperItem from '../src/StepperItem'
import { StepperProps } from '../src/types'

const defaultSlots = [
  h(IxStepperItem, { key: 1, title: 'Finished', description: 'This is a description.' }),
  h(IxStepperItem, { key: 2, title: 'In Progress', description: 'This is a description.' }),
  h(IxStepperItem, { key: 3, title: 'Waiting', description: 'This is a description.' }),
]

describe('Stepper', () => {
  const StepperMount = (options?: MountingOptions<Partial<StepperProps>>) => {
    const { props, slots, ...rest } = options || {}
    return mount(IxStepper, {
      ...rest,
      props: { activeKey: 2, ...props },
      slots: { default: () => [...defaultSlots], ...slots },
    })
  }

  renderWork<StepperProps>(IxStepper, { props: { activeKey: 2 }, slots: { default: () => defaultSlots } })

  test('active work', async () => {
    const onUpdateActiveKey = jest.fn()
    const wrapper = StepperMount({ props: { activeKey: 1, 'onUpdate:activeKey': onUpdateActiveKey } })

    expect(wrapper.findAll('.ix-stepper-item')[0].classes()).toContain('ix-stepper-item-active')

    await wrapper.setProps({ activeKey: 2 })

    expect(wrapper.findAll('.ix-stepper-item')[0].classes()).not.toContain('ix-stepper-item-active')
    expect(wrapper.findAll('.ix-stepper-item')[1].classes()).toContain('ix-stepper-item-active')
  })

  test('clickable work', async () => {
    const onUpdateActiveKey = jest.fn()
    const wrapper = StepperMount({ props: { activeKey: 1, clickable: true, 'onUpdate:activeKey': onUpdateActiveKey } })

    expect(wrapper.findAll('.ix-stepper-item')[0].classes()).toContain('ix-stepper-item-clickable')
    expect(wrapper.findAll('.ix-stepper-item')[0].classes()).toContain('ix-stepper-item-active')

    await wrapper.findAll('.ix-stepper-item')[1].trigger('click')

    expect(onUpdateActiveKey).toBeCalledTimes(1)
    expect(onUpdateActiveKey).toBeCalledWith(2)

    await wrapper.setProps({ activeKey: 2 })
    await wrapper.findAll('.ix-stepper-item')[1].trigger('click')

    expect(onUpdateActiveKey).toBeCalledTimes(1)
  })

  test('labelPlacement work', async () => {
    const wrapper = StepperMount({ props: { labelPlacement: 'bottom' } })

    expect(wrapper.classes()).toContain('ix-stepper-label-bottom')

    await wrapper.setProps({ labelPlacement: 'end' })

    expect(wrapper.classes()).toContain('ix-stepper-label-end')
  })

  test('percent work', async () => {
    const wrapper = StepperMount({ props: { percent: 10 } })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ percent: 50 })

    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ percent: 100 })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('size work', async () => {
    const wrapper = StepperMount({ props: { size: 'sm' } })

    expect(wrapper.classes()).toContain('ix-stepper-sm')

    await wrapper.setProps({ size: 'md' })

    expect(wrapper.classes()).toContain('ix-stepper-md')
  })

  test('status work', async () => {
    const wrapper = StepperMount({ props: { status: 'process' } })

    let items = wrapper.findAll('.ix-stepper-item')

    expect(items[0].classes()).toContain('ix-stepper-item-finish')
    expect(items[1].classes()).toContain('ix-stepper-item-process')
    expect(items[2].classes()).toContain('ix-stepper-item-wait')

    await wrapper.setProps({ status: 'finish' })

    items = wrapper.findAll('.ix-stepper-item')

    expect(items[0].classes()).toContain('ix-stepper-item-finish')
    expect(items[1].classes()).toContain('ix-stepper-item-finish')
    expect(items[2].classes()).toContain('ix-stepper-item-wait')

    await wrapper.setProps({ status: 'error' })

    expect(items[0].classes()).toContain('ix-stepper-item-finish')
    expect(items[1].classes()).toContain('ix-stepper-item-error')
    expect(items[2].classes()).toContain('ix-stepper-item-wait')
  })
})
