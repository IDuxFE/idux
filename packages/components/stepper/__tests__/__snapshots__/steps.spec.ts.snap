// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper percent work 1`] = `
"<div class=\\"ix-stepper ix-stepper-label-end ix-stepper-md ix-stepper-with-percent\\">
  <div class=\\"ix-stepper-item ix-stepper-item-finish\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><i class=\\"ix-icon ix-icon-check\\" role=\\"img\\" aria-label=\\"check\\"></i></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Finished</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-process ix-stepper-item-active\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\">
      <div class=\\"ix-progress-circle ix-progress ix-progress-normal\\" style=\\"width: 40px; height: 40px; font-size: 12px;\\"><svg viewBox=\\"0 0 100 100\\">
          <!---->
          <path class=\\"ix-progress-circle-trail\\" style=\\"stroke-dasharray: 301.59289474462014px 301.59289474462014px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
          <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 30.159289474462014px 301.59289474462014px; stroke-dashoffset: -0px;\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
        </svg>
        <div class=\\"ix-progress-info\\">
          <!---->
        </div>
      </div><span>2</span>
    </div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">In Progress</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-wait\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><span>3</span></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Waiting</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
</div>"
`;

exports[`Stepper percent work 2`] = `
"<div class=\\"ix-stepper ix-stepper-label-end ix-stepper-md ix-stepper-with-percent\\">
  <div class=\\"ix-stepper-item ix-stepper-item-finish\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><i class=\\"ix-icon ix-icon-check\\" role=\\"img\\" aria-label=\\"check\\"></i></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Finished</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-process ix-stepper-item-active\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\">
      <div class=\\"ix-progress-circle ix-progress ix-progress-normal\\" style=\\"width: 40px; height: 40px; font-size: 12px;\\"><svg viewBox=\\"0 0 100 100\\">
          <!---->
          <path class=\\"ix-progress-circle-trail\\" style=\\"stroke-dasharray: 301.59289474462014px 301.59289474462014px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
          <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 150.79644737231007px 301.59289474462014px; stroke-dashoffset: -0px;\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
        </svg>
        <div class=\\"ix-progress-info\\">
          <!---->
        </div>
      </div><span>2</span>
    </div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">In Progress</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-wait\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><span>3</span></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Waiting</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
</div>"
`;

exports[`Stepper percent work 3`] = `
"<div class=\\"ix-stepper ix-stepper-label-end ix-stepper-md ix-stepper-with-percent\\">
  <div class=\\"ix-stepper-item ix-stepper-item-finish\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><i class=\\"ix-icon ix-icon-check\\" role=\\"img\\" aria-label=\\"check\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" aria-hidden=\\"true\\" data-icon=\\"check\\">
          <path d=\\"M167.29 445.278 437.84 715.83a10.779 10.779 0 0 0 15.2 0l403.886-403.78a10.779 10.779 0 0 1 15.253 0l22.85 22.852a10.779 10.779 0 0 1 0 15.252L453.094 792.091a10.779 10.779 0 0 1-15.252 0L129.13 483.38a10.779 10.779 0 0 1 0-15.251l22.906-22.852a10.779 10.779 0 0 1 15.252 0z\\"></path>
        </svg></i></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Finished</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-process ix-stepper-item-active\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\">
      <div class=\\"ix-progress-circle ix-progress ix-progress-success\\" style=\\"width: 40px; height: 40px; font-size: 12px;\\"><svg viewBox=\\"0 0 100 100\\">
          <!---->
          <path class=\\"ix-progress-circle-trail\\" style=\\"stroke-dasharray: 301.59289474462014px 301.59289474462014px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;\\" stroke=\\"#f5f5f5\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
          <path class=\\"ix-progress-circle-path ix-progress-circle-bg\\" style=\\"transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s; stroke-dasharray: 301.59289474462014px 301.59289474462014px; stroke-dashoffset: -0px;\\" fill-opacity=\\"0\\" stroke-linecap=\\"round\\" stroke-width=\\"4\\" d=\\"M 50,50 m 0,-48
       a 48,48 0 1 1 0,96
       a 48,48 0 1 1 0,-96\\"></path>
        </svg>
        <div class=\\"ix-progress-info\\">
          <!---->
        </div>
      </div><span>2</span>
    </div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">In Progress</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-wait\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><span>3</span></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Waiting</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
</div>"
`;

exports[`Stepper render work 1`] = `
"<div class=\\"ix-stepper ix-stepper-label-end ix-stepper-md\\">
  <div class=\\"ix-stepper-item ix-stepper-item-finish\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><i class=\\"ix-icon ix-icon-check\\" role=\\"img\\" aria-label=\\"check\\"></i></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Finished</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-process ix-stepper-item-active\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><span>2</span></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">In Progress</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
  <div class=\\"ix-stepper-item ix-stepper-item-wait\\">
    <div class=\\"ix-stepper-item-tail\\"></div>
    <div class=\\"ix-stepper-item-icon\\"><span>3</span></div>
    <div class=\\"ix-stepper-item-content\\">
      <div class=\\"ix-stepper-item-title\\">Waiting</div>
      <div class=\\"ix-stepper-item-description\\">This is a description.</div>
    </div>
  </div>
</div>"
`;
